test/scan_by_key.cpp

```diff
@@ -106,7 +106,7 @@ void verify(af::dim4 dims,
                     gold = (To)in[index];
                     ASSERT_NEAR(gold, out[index], eps);
                 } else {
-                    gold = binOp.init();
+                    gold = Binary<To, op>::init();
```

Comment:
Binary is not a public facing API and should not be used in test. Please remove it.

Reply:
So I think this test is using a separate header file that defines its own templated Binary operators (test/binary_ops.hpp), apart from the ArrayFire source. I accidentally made this static too in the course of scanning the whole repo for all Binary operator implementations. Let me just revert this and test/binary_ops.hpp to it's original version instead then?